src/monattack.cpp

```diff
@@ -4860,3 +4860,175 @@ bool mattack::dodge_check(monster *z, Creature *target){
     }
     return false;
 }
+
+void mattack::ink_jet(monster *z, int index) {
+
+    if (!z->can_act()) {
+        return;
+    }
+
+    int jet_range =  5;
+
+    // TODO: make the creature utilise this ability then escaping the threat
+    Creature *target = z->attack_target();
+
+    if (target == nullptr || rl_dist(z->pos(), target->pos()) > jet_range || !z->sees(*target)) {
+        return;
+    }
+
+    std::vector<tripoint> line = g->m.find_clear_path(z->pos(), target->pos());
+
+    z->reset_special(index);
+
+    z->moves -= 50;
+
+    const game_message_type msg_type = target->attitude_to(g->u) == Creature::A_FRIENDLY ? m_bad : m_warning;
+
+    // Notify the player, if he sees the use of this ability
+    bool u_see = g->u.sees(*z);
+
+    if (u_see && !target->is_monster()) {
+        target->add_msg_player_or_npc(msg_type, 
+            _("The %s squirts a jet of black ink at you!"),
+            _("The %s squirts a jet of black ink at <npcname>!"),
+            z->name().c_str());
+    }
+    else if (u_see) {
+        add_msg(msg_type, 
+            _("The %1$s squirts a jet of black ink at %2$s!"),
+            z->name().c_str(), target->disp_name().c_str());
+    }
+
+    // Shoot the jet of ink
+
+    for (auto &i : line) {
+        g->m.add_field(i, fd_ink, 1, 0);
+        if (g->m.move_cost(i) == 0) {
+            g->m.add_field(i, fd_ink, 3, 0);
+            if (g->u.sees(i))
+                add_msg(_("The ink splatters on the %s!"),
+                    g->m.tername(i).c_str());
+            return;
+        }
+    }
+
+    // Handle the effects of creature, potentially, getting hit by the ink
+    // TODO: make it do something then affecting non player/npc target
+    // TODO: if you can't see, you can't dodge this effect
+    if (!target->uncanny_dodge()) {
+        if (rng(0, 10) > target->get_dodge() || one_in(target->get_dodge())) {
+            target->on_dodge(z, 10);
+            if ( one_in(5)) {
+                add_msg(m_bad, _("The ink jet hits your eyes!"));
+                target->add_env_effect( "ooze_in_eyes", bp_eyes, 5, 30 );
+                target->add_env_effect("inked", bp_torso, 4, 50);
+            }
+            else {
+                target->add_env_effect( "inked", bp_torso, 4, 150 );
+            }
+        }
+        else if (u_see && !target->is_monster()) {
```

Comment:
You don't need a separate branch for monsters. `add_msg_player_or_npc` can handle monsters just fine (it substitutes monster name for `<npcname>`).